### PR TITLE
chore: release prep for v1.42.0

### PR DESCRIPTION
**Pull request by Prashansa-K**
- Updated go-database-reconciler to v1.17.0
- Added changelog

### CODE REVIEW

**Review by AntoineJac — COMMENTED** [review]
LGTM, ok to merge.

File: CHANGELOG.md [hunk 1]
```diff
@@ -101,6 +102,36 @@
 - [v0.2.0](#v020)
 - [v0.1.0](#v010)
 
+## [v1.42.0]
```
Comment (pmalek) [inline CHANGELOG.md:105]:
```suggestion
## [v1.42.0]

```

File: CHANGELOG.md [hunk 2]
```diff
@@ -101,6 +102,36 @@
...
+deck dump, sync and diff. This is valid for Konnect entities only.
+[#1478](https://github.com/Kong/deck/pull/1478)
+[go-database-reconciler #153](https://github.com/Kong/go-database-reconciler/pull/153)
+- Improved speed for deck sync/diff operations involving consumer-groups 
+for gw 3.9+. The underlying API call to `GET /consumer_group` is called
+with query parameter `list_consumers=false`, making it faster for deck
+to deal with cases where a consumer-group holds many consumers.
+(#1475)[https://github.com/Kong/deck/pull/1475]
+(go-kong #487)[https://github.com/Kong/go-kong/pull/487]
+
```
Comment (pmalek) [inline CHANGELOG.md:122]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -101,6 +102,36 @@
...
+
+### Fixes
```
Comment (pmalek) [inline CHANGELOG.md:124]:
```suggestion
### Fixes

```

**Review by AntoineJac — APPROVED** [review]

**Pull request merged**